### ISSUE TITLE
"ReferenceError: Buffer is not defined" with v5.1.1

### ISSUE THREAD

**Issue by Sirz3chs** [node-cache/node-cache#197]
Actually, i used the lib in the prvious version (v5.1.0) and it works great. Since i updated to the v5.1.1 i have the error below and don't know if i missed some configuration to handle this.

![image](https://user-images.githubusercontent.com/23245718/85845680-ec961600-b7a4-11ea-9219-8fce0fd1feee.png)

This is how i define my cache:
```javascript
import * as NodeCache from "node-cache";

const FILTERS_CACHE_TTL = 300;

const FILTERS_CACHE_CLEAN_PERIOD = 60;

private readonly filtersCache = new NodeCache({
    stdTTL: FILTERS_CACHE_TTL,
    checkperiod: FILTERS_CACHE_CLEAN_PERIOD
});
```

**Comment by erdii** [comment]
Hey @Sirz3chs welcome :)

I wonder if you are using node-cache in a browser environment since `Buffer` is not available there?

**Comment by jdussouillez** [comment]
Yes we are using this library in a browser (Angular app). _I'm @Sirz3chs colleague._

The commit https://github.com/node-cache/node-cache/commit/ec5c5ad4edce7481a902201f3fd1c14327083a18 broke the `set` function for us when we upgraded to v5.1.1. We had to revert to v5.1.0.

I tried to create a [Stackblitz](https://stackblitz.com/edit/angular-nodecache-issue-197?file=src%2Fapp%2Fapp.component.ts) to reproduce it but it's working in it. I'll take some time this week to go deeper

**Comment by erdii** [comment]
Thanks for pointing out! I see!

I fixed it now and let it test for `Buffer` availability first.

The thing is: node-cache was not written for the browser, only node.js and we also depend on other things not available in regular browsers:
- `require( "clone" ) - npm package
- `require('events').EventEmitter` - internal node.js package

I am glad that you are able to run it in the browser and would be interested in getting to know your bundling setup.
Maybe we can then add browser support to our test matrix :blush:

**Comment by jdussouillez** [comment]
> i will release a new bugfix version as soon the fix-pr has been reviewed

There is no emergency for this. We chose to revert to the v5.1.0 but we could have use other solutions (see the details below) but your fix is the better way to handle it.

Anyway, thanks a lot for the fix !

----
Workaround for people like us who uses node-cache in a Angular web app and don't want/can't downgrade from v5.1.1 : Install the [buffer](https://www.npmjs.com/package/buffer) package and add this line in the `polyfill.ts` :

```typescript
// Fix the "Buffer is not defined" error after upgrading to node-cache v5.1.1
// See https://github.com/node-cache/node-cache/issues/197
(window as any).global = window;
window.Buffer = window.Buffer || require("buffer").Buffer;
```

**Comment by erdii** [comment]
the fixed version v5.1.2 has been released so there should be no need to polyfill `Buffer` anymore :tada: 

---

Ah I see. Are you also polifilling `require("events")` somewhere?
Thanks for the insight!

**Comment by jdussouillez** [comment]
Thanks for the quick fix !

----

Here's the "node-cache" section in my `package-lock.json` :
```json
"node-cache": {
  "version": "5.1.0",
  "resolved": "https://registry.npmjs.org/node-cache/-/node-cache-5.1.0.tgz",
  "integrity": "sha512-gFQwYdoOztBuPlwg6DKQEf50G+gkK69aqLnw4djkmlHCzeVrLJfwvg9xl4RCAGviTIMUVoqcyoZ/V/wPEu/VVg==",
  "requires": {
    "clone": "2.x"
  }
},
```

The `events` library is also part of my `package-lock.json` because it's required by the `node-libs-browser` package which is required by `webpack` which is required by `@angular-devkit/build-angular`. I had nothing to do in the polyfill file to make "node-cache" work.

My current polyfill file for v5.1.0 (generated by Angular, except the last line) :

```typescript
/***************************************************************************************************
 * Zone JS is required by default for Angular itself.
 */
import "zone.js/dist/zone";  // Included with Angular CLI.

/******************************************************************
 * Load `$localize` - used if i18n tags appear in Angular templates.
 */
// https://angular.io/guide/migration-localize
import "@angular/localize/init";

/***************************************************************************************************
 * APPLICATION IMPORTS
 */
(window as any).global = window; // This is needed for another library
```

I hope this helps you.
